identity/README.md

```diff
@@ -0,0 +1,279 @@
+  returns (GetIdentityResponse) {}
+
+  // GetCapabilities returns the capabilities that the CSI-driver supports.
+  rpc GetCapabilities(GetCapabilitiesRequest)
+  returns (GetCapabilitiesResponse) {}
+
+  // Probe is called by the CO plugin to validate that the CSI-Addons Node is
+  // still healthy.
+  rpc Probe(ProbeRequest)
+  returns (ProbeResponse) {}
+}
+```
+
+### `GetIdentity`
+
+The CSI-driver needs to provide its driver-name so that the CSI-Addons CO
+plugin can invoke RPCs for volumes that use this driver. There is an option for
+the CSI-driver to provide details about its version and a key/value manifest.
+
+```protobuf
+// GetIdentityRequest is sent by the CSI-Addons CO plugin to obtain the
+// drivername, version and optional details from the CSI-driver.
+message GetIdentityRequest {
+  // Intentionally empty.
+}
+
+// GetIdentityResponse is returned by the CSI-driver as a response to a
+// GetIdentityRequest.
+message GetIdentityResponse {
+  // The name MUST follow domain name notation format
+  // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD include
+  // the CSI-drivers's host company name and the CSI-driver name, to minimize
+  // the possibility of collisions. It MUST be 63 characters or less, beginning
+  // and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-),
+  // dots (.), and alphanumerics between. This field is REQUIRED.
+  string name = 1;
+  // This field is REQUIRED. Value of this field is opaque to the CO.
+  string vendor_version = 2;
+  // This field is OPTIONAL. Values are opaque to the CO.
+  map<string, string> manifest = 3;
+}
+```
+
+#### GetIdentity Errors
+
+If the CSI-driver is unable to complete the GetIdentity call successfully, it
+MUST return a non-ok gRPC code in the gRPC status.
+
+### `GetCapabilities`
+
+The CSI-Addons plugin for the CO needs to detect the supported features for
+each CSI-driver. Features are described as capabilities, each with additional
+information on how the CSI-Addons CO plugin may invoke the operation.
+
+There might be operations that need to run on the CSI Controller (provisioner)
+for storage system access. Other operations may require running on a volume
+that is attached/staged on a node. By specifying the type of Service that the
+CSI-driver provides, the CSI-Addons CO plugin can make informed decisions on
+where to invoke the RPCs for an operation.
+
+```protobuf
+// GetCapabilitiesRequest is sent by the CSI-Addons CO plugin to detect the
+// features that a CSI-driver supports.
+message GetCapabilitiesRequest {
+  // Intentionally empty.
+}
+
+// GetCapabilitiesResponse is returned by the CSI-driver as a response to a
+// GetCapabilitiesRequest.
+message GetCapabilitiesResponse {
+  // All the capabilities that the controller service supports. This
+  // field is OPTIONAL.
+  repeated Capability capabilities = 1;
+}
+
+// Specifies one or more capabilities of the CSI-driver.
+message Capability {
```

Comment:
One thing which is not clear to me here is, can a csi driver advertise only  one capability (ex; "node"). Is it allowed by spec?

Reply:
Sure, a CSI-driver can provide `NODE_SERVICE` only. It is pretty useless though, ideally some capabilities to run some operations (ReclaimSpace) would be supported by the CSI-driver too.

Reply:
we never know, that said, there are CSI drivers without controller capabilities,, so good to know it is flexible to either of it or both...etc :+1: